src/python/pants/core/goals/repl.py

```diff
@@ -101,20 +99,17 @@ async def run_repl(
         )
         return Repl(-1)
 
-    repl_impl = repl_implementation_cls(
-        targets=Targets(
-            tgt for tgt in transitive_targets.closure if repl_implementation_cls.is_valid(tgt)
-        )
-    )
+    repl_impl = repl_implementation_cls(targets=Targets(transitive_targets.closure))
     request = await Get(ReplRequest, ReplImplementation, repl_impl)
 
     with temporary_dir(root_dir=global_options.options.pants_workdir, cleanup=False) as tmpdir:
         tmpdir_relative_path = PurePath(tmpdir).relative_to(build_root.path).as_posix()
         exe_path = PurePath(tmpdir, request.binary_name).as_posix()
         workspace.write_digest(request.digest, path_prefix=tmpdir_relative_path)
```

Comment:
I'm not sure why we do this. Instead, I think that we could use `InteractiveProcess(input_digest=request.digest)` and don't set `run_in_workspace=True`. This would allow us to remove the tmpdir and pushd.

Reply:
Okay, yes, confirmed that this works:

```python
    result = interactive_runner.run(
        InteractiveProcess(
            argv=(request.binary_name,), input_digest=request.digest, env=request.env
        )
    )
```

This allows us to delete a lot of code, including several Params to the goal_rule.

Reply:
Neat! Love it.